source/dem/floating_grid.cc

```diff
@@ -0,0 +1,64 @@
+//
+// Created by Emile Bergeron on 2022-01-30.
+//
+#include "dem/floating_grid.h"
+
+#include <core/solutions_output.h>
+
+#include "dem/read_mesh.h"
+
+#include <deal.II/grid/grid_out.h>
+
+template <int dim, int spacedim>
+FloatingGrid<dim, spacedim>::FloatingGrid(
+  const DEMSolverParameters<spacedim> &dem_parameters,
+  const ConditionalOStream &           pcout,
+  const double &                       dem_time_step)
+  : gridMotion(dem_parameters, dem_time_step)
+{
+  if (dem_parameters.floating_grid.mesh.file_name != "none")
```

Comment:
It would be better to have a number of floating grids parameter which is set to 0, this will pave the support for multiple one instead of using a string for that.

Reply:
I changed the code so that the FloatingGrid class only takes a subsection of the parameters as an arugument, which should make this change easy (only need to change the parser, more or less). I will implement this in the next PR as I add the code to the DEM solver.